mesh/nurbs.cpp

```diff
@@ -341,6 +341,100 @@ void KnotVector::CalcDnShape(Vector &gradn, int n, int i, double xi) const
 
 }
 
+void KnotVector::FindMaxima(Array<int> &ks,
+                            Vector &xi,
+                            Vector &u)
+{
+   Vector shape(Order+1);
+   Vector maxima(GetNCP());
+   double arg1,arg2,arg,max1,max2,max;
+
+   xi.SetSize(GetNCP());
+   u.SetSize(GetNCP());
+   ks.SetSize(GetNCP());
+   for (int j = 0; j <GetNCP(); j++)
+   {
+      maxima[j] = 0;
+      for (int d = 0; d < Order+1; d++)
+      {
+         int i = j - d;
+         if (isElement(i))
+         {
+            arg1 = 0;
+            CalcShape ( shape, i, arg1);
+            max1 = shape[d];
+
+            arg2 = 1;
+            CalcShape ( shape, i, arg2);
+            max2 = shape[d];
+
+            arg = (arg1 + arg2)/2;
+            CalcShape ( shape, i, arg);
+            max = shape[d];
+
+            while ( ( max > max1 ) || (max > max2) )
```

Comment:
In the miniapp example, this search takes about 50 iterations. Newton's method, restricted to the unit interval, could solve this much faster. Since it is such a small problem, I suppose it will not hurt performance. So I am fine with leaving this as it is, but you can add the Newton solver if you are interested.

Reply:
See the comments in a review below: [link](https://github.com/mfem/mfem/pull/3232#discussion_r1008495962)